### PR TITLE
Zip-Export: Split by layers option

### PR DESCRIPTION
**Pull request by MaxEden**
Added option to export non-merged frames

### CODE REVIEW

**Comment by juliandescottes** [comment]
Hi, 

Thanks for the PR, I'll have a look later today !

**Comment by juliandescottes** [comment]
@MaxEden Can you give me a bit of context here ? How do you take advantage of this splitted export in your workflow ?

Thanks !

**Comment by MaxEden** [comment]
I draw sprites for games, and the result sprites often need to be made of several separate parts, for example: legs/torso/arms/weapon. It's easy to animate the pieces together, but I still need them separated so I can combine them dynamically or apply sfx only on some parts.
https://www.dropbox.com/s/mgbce3ztho2nsug/Squirrel.piskel

**Comment by juliandescottes** [comment]
Ok ! Thanks for info, seems reasonable :)

File: src/js/controller/settings/exportimage/PngExportController.js [hunk 1]
```diff
@@ -30,12 +32,34 @@
   ns.PngExportController.prototype.onZipButtonClick_ = function () {
     var zip = new window.JSZip();
 
-    for (var i = 0 ; i < this.piskelController.getFrameCount() ; i++) {
-      var frame = this.piskelController.getFrameAt(i);
-      var canvas = this.getFrameAsCanvas_(frame);
-      var basename = this.pngFilePrefixInput.value;
-      var filename =  basename + (i + 1) + '.png';
-      zip.file(filename, pskl.utils.CanvasUtils.getBase64FromCanvas(canvas) + '\n', {base64: true});
+    function splittedExport() {
```
Comment (juliandescottes) [inline src/js/controller/settings/exportimage/PngExportController.js]: We don't normally use named functions in Piskel, but try to attach functions to prototypes (except for some anonymous functions in forEach constructs, callbacks and so on ...).

To be consistent, we should do the same here.

File: src/js/controller/settings/exportimage/PngExportController.js [hunk 2]
```diff
@@ -30,12 +32,34 @@
...
+          var basename = this.pngFilePrefixInput.value;
+          var filename = 'l' + j + '_' + basename + (i + 1) + '.png';
+          zip.file(filename, pskl.utils.CanvasUtils.getBase64FromCanvas(canvas) + '\n', {base64: true});
+        }
+      }
+    }
+
+    function mergedExport() {
+      for (var i = 0; i < this.piskelController.getFrameCount(); i++) {
+        var frame = this.piskelController.getFrameAt(i);
```
Comment (juliandescottes) [inline src/js/controller/settings/exportimage/PngExportController.js]: We could mutualize some code between mergedExport and splittedExport ?

Comment (MaxEden) [inline src/js/controller/settings/exportimage/PngExportController.js]: Maybe, but not much without drenching it with ifs. Perhaps I don't see how.

File: src/js/controller/settings/exportimage/PngExportController.js (hunk 1 above)
Comment (MaxEden) [inline src/js/controller/settings/exportimage/PngExportController.js]: Something like this?
http://pastebin.com/gGZCmAyc

Comment (juliandescottes) [inline src/js/controller/settings/exportimage/PngExportController.js]: Yes , that'd be great

File: src/js/controller/settings/exportimage/PngExportController.js (hunk 2 above)
Comment (juliandescottes) [inline src/js/controller/settings/exportimage/PngExportController.js]: Fine with keeping this as it is.

**Comment by juliandescottes** [comment]
@MaxEden : if you can update your PR with the change you linked on pastebin, I'll merge it right after.

Thanks !

**Comment by MaxEden** [comment]
Updated btw

**Comment by juliandescottes** [comment]
Thanks for the ping, didn't see that.

**Pull request merged**